Node/samples/app.js

```diff
@@ -145,13 +154,61 @@ bot.dialog('StartNewReplyChain', function (session) {
 });
 bot.dialog('MentionUser', function (session) {
     // user name/user id
-    var toMention = {
-        name: 'Bill Zeng',
-        id: userId
+    var user = {
+        id: userId,
+        name: 'Bill Zeng'
     };
-    var msg = new teams.TeamsMessage(session).text(teams.TeamsMessage.getTenantId(session.message));
-    var mentionedMsg = msg.addMentionToText(toMention);
-    session.send(mentionedMsg);
+    var mention = new teams.UserMention(user);
+    var msg = new teams.TeamsMessage(session).addEntity(mention).text(mention.text + ' ' + teams.TeamsMessage.getTenantId(session.message));
+    session.send(msg);
+    session.endDialog();
+});
+bot.dialog('MentionChannel', function (session) {
+    // user name/user id
+    var channelId = null;
+    if (session.message.address.conversation.id) {
+        var splitted = session.message.address.conversation.id.split(';', 1);
+        channelId = splitted[0];
+    }
+    var teamId = session.message.sourceEvent.team.id;
+    connector.fetchChannelList(session.message.address.serviceUrl, teamId, function (err, result) {
+        if (err) {
+            session.endDialog('There is some error');
+        }
+        else {
+            var channelName = null;
+            for (var i in result) {
+                var channelInfo = result[i];
+                if (channelId == channelInfo['id']) {
+                    channelName = channelInfo['name'] || 'General';
+                    break;
+                }
+            }
+            var channel = {
+                id: channelId,
+                name: channelName
+            };
+            var mention = new teams.ChannelMention(channel);
+            var msg = new teams.TeamsMessage(session).addEntity(mention).text('This is a test message to at mention the channel. ' + mention.text);
+            session.send(msg);
+            session.endDialog();
+        }
+    });
+});
+bot.dialog('MentionTeam', function (session) {
+    // user name/user id
+    var channelId = null;
+    if (session.message.address.conversation.id) {
+        var splitted = session.message.address.conversation.id.split(';', 1);
+        channelId = splitted[0];
+    }
+    var team = {
+        id: channelId,
+        name: 'All'
```

Comment:
Could you please also add example of how to retrieve the team name similarly as you did for channel name? I think there is an API to GetTeamProfile to fetch that info.